modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -0,0 +1,161 @@
+
+  //Set up domain bounds with mesh tools
+  for (unsigned int i = 0; i < LIBMESH_DIM; i++)
+  {
+    _bottom_left(i) = _mesh.getMinInDimension(i);
+    _top_right(i) = _mesh.getMaxInDimension(i);
+  }
+  _range = _top_right - _bottom_left;
+
+  if (_semiaxis_a_variation > 0.0 && _semiaxis_variation_type == 2)
+    mooseError("If Semiaxis_a_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
+  if (_semiaxis_b_variation > 0.0 && _semiaxis_variation_type == 2)
+    mooseError("If Semiaxis_b_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
+  if (_semiaxis_c_variation > 0.0 && _semiaxis_variation_type == 2)
+    mooseError("If Semiaxis_c_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
+
+  SmoothSuperellipsoidBaseIC::initialSetup();
+}
+
+void
+MultiSmoothSuperellipsoidIC::computeSuperellipsoidSemiaxes()
+{
+  _as.resize(_numbub);
+  _bs.resize(_numbub);
+  _cs.resize(_numbub);
+
+  for (unsigned int i = 0; i < _numbub; i++)
+  {
+    //Vary bubble radius
+    switch (_semiaxis_variation_type)
+    {
+    case 0: //Uniform distrubtion
+      _as[i] = _semiaxis_a*(1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_a_variation);
+      _bs[i] = _semiaxis_b*(1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_b_variation);
+      _cs[i] = _semiaxis_c*(1.0 + (1.0 - 2.0 * _random.rand(_tid)) * _semiaxis_c_variation);
+      break;
+    case 1: //Normal distribution
+      _as[i] = _random.randNormal(_tid, _semiaxis_a,_semiaxis_a_variation);
+      _bs[i] = _random.randNormal(_tid, _semiaxis_b,_semiaxis_b_variation);
+      _cs[i] = _random.randNormal(_tid, _semiaxis_c,_semiaxis_c_variation);
+      break;
+    case 2: //No variation
+      _as[i] = _semiaxis_a;
+      _bs[i] = _semiaxis_b;
+      _cs[i] = _semiaxis_c;
+    }
+
+    if (_as[i] < 0.0) _as[i] = 0.0;
+    if (_bs[i] < 0.0) _bs[i] = 0.0;
+    if (_cs[i] < 0.0) _cs[i] = 0.0;
+  }
+}
+
+
+void
+MultiSmoothSuperellipsoidIC::computeSuperellipsoidCenters()
+{
+  _centers.resize(_numbub);
+
+  for (unsigned int i = 0; i < _numbub; i++)
+  {
+    //Vary Superellipsoid center positions
+    unsigned int num_tries = 0;
+
+    Real rr = 0.0;
+    Point newcenter = 0.0;
+
+    while (rr < _bubspac && num_tries < _numtries)
+    {
+      num_tries++;
+      //Moose::out<<"num_tries: "<<num_tries<<std::endl;
+
+      Real ran1 = _random.rand(_tid);
+      Real ran2 = _random.rand(_tid);
+      Real ran3 = _random.rand(_tid);
+
+      newcenter(0) = _bottom_left(0) + ran1*_range(0);
+      newcenter(1) = _bottom_left(1) + ran2*_range(1);
+      newcenter(2) = _bottom_left(2) + ran3*_range(2);
+
+      for (unsigned int j = 0; j < i; j++)
+      {
+        if (j == 0) rr = _range.norm();
+
+        Real tmp_rr = _mesh.minPeriodicDistance(_var.number(), _centers[j], newcenter);
+
+        if (tmp_rr < rr) rr = tmp_rr;
+      }
+
+      if (i == 0) rr = _range.norm();
+    }
```

Comment:
interesting... It seems like this code would be useful in other ICs. We have the same problem with assigning centers in the VoronoiIC as well. Are you interested in putting this code into the PhaseField utils? If not that's fine. somebody else can move it later.